### ISSUE TITLE
Error encountered when setting maximum gap opens to 2

### ISSUE THREAD

**Issue by KrisTabb** [MikkelSchubert/paleomix#57]
I've been using Paleomix to process some ancient DNA sequence reads and it has been a tremendous help for my project. However, I recently tried to optimise my BWA backtrack alignment parameters and came across an obstacle: if I run the pipeline with the default maximum number of gap opens [1], the pipeline runs without any issues, but if I increase this to 2, I get the error below and the pipeline does not run to completion. All other parameters are the same, including -n: 0.01 and MinQuality: 30 and UseSeed: no. Can any of these settings be incompatible with the other and cause the error below? Thanks in advance for letting me know what you think!

 2024-09-20 15:25:22,695 paleomix.pipeline ERROR NodeError while alignment of './GD_Boncuklu_mice_reanalysis_linear_o2/reads/KT_LB_140324_A1/KT_LB_140324_A1/Lane_1/KT_LB_140324_A1_EKDL240005610-1A_22KL5WLT3_L8_x.fq/reads.collapsed.gz' onto GCF_000001635.27_GRCm39_genomic using BWA samse:
2024-09-20 15:25:22,696 paleomix.pipeline ERROR     Error(s) running Node:
2024-09-20 15:25:22,696 paleomix.pipeline ERROR         Temporary directory: '/media/jbod4/kristina_tabbada_jbod4/mus_merged_data/temp/fdc978d4-0eb1-4f1c-918f-45b159d50d09'
2024-09-20 15:25:22,696 paleomix.pipeline ERROR
2024-09-20 15:25:22,697 paleomix.pipeline ERROR     Parallel processes:
2024-09-20 15:25:22,697 paleomix.pipeline ERROR       Process 1:
2024-09-20 15:25:22,697 paleomix.pipeline ERROR         Command = bwa samse -n 0.01 -o 2 \
2024-09-20 15:25:22,697 paleomix.pipeline ERROR                       /media/jbod2/kristina_jbod2/Genomes/Mus_musculus_dom/GCF_000001635.27_GRCm39_genomic.fasta \
2024-09-20 15:25:22,697 paleomix.pipeline ERROR                       ./GD_Boncuklu_mice_reanalysis_linear_o2/Mus_musculus_dom/KT_LB_140324_A1/KT_LB_140324_A1/Lane_1/KT_LB_140324_A1_EKDL240005610-1A_22KL5WLT3_L8_x.fq/collapsed.sai \
2024-09-20 15:25:22,697 paleomix.pipeline ERROR                       ./GD_Boncuklu_mice_reanalysis_linear_o2/reads/KT_LB_140324_A1/KT_LB_140324_A1/Lane_1/KT_LB_140324_A1_EKDL240005610-1A_22KL5WLT3_L8_x.fq/reads.collapsed.gz
2024-09-20 15:25:22,697 paleomix.pipeline ERROR         Status  = Exited with return-code 1
2024-09-20 15:25:22,698 paleomix.pipeline ERROR         STDOUT  = Piped to process 2
2024-09-20 15:25:22,698 paleomix.pipeline ERROR         STDERR* = '/media/jbod4/kristina_tabbada_jbod4/mus_merged_data/temp/fdc978d4-0eb1-4f1c-918f-45b159d50d09/pipe_bwa_140447126229920.stderr'
2024-09-20 15:25:22,698 paleomix.pipeline ERROR         CWD     = '/media/jbod4/kristina_tabbada_jbod4/mus_merged_data'
2024-09-20 15:25:22,698 paleomix.pipeline ERROR
2024-09-20 15:25:22,698 paleomix.pipeline ERROR       Process 2:
2024-09-20 15:25:22,698 paleomix.pipeline ERROR         Command = /home/kristina/miniconda3/envs/paleomix/bin/python \
2024-09-20 15:25:22,698 paleomix.pipeline ERROR                       /home/kristina/miniconda3/envs/paleomix/lib/python3.9/site-packages/paleomix/main.py \
2024-09-20 15:25:22,699 paleomix.pipeline ERROR                       cleanup --fasta \
2024-09-20 15:25:22,699 paleomix.pipeline ERROR                       /media/jbod2/kristina_jbod2/Genomes/Mus_musculus_dom/GCF_000001635.27_GRCm39_genomic.fasta \
2024-09-20 15:25:22,699 paleomix.pipeline ERROR                       --temp-prefix \
2024-09-20 15:25:22,699 paleomix.pipeline ERROR                       /media/jbod4/kristina_tabbada_jbod4/mus_merged_data/temp/fdc978d4-0eb1-4f1c-918f-45b159d50d09/bam_cleanup \
2024-09-20 15:25:22,699 paleomix.pipeline ERROR                       --rg-id KT_LB_140324_A1 --rg SM:KT_LB_140324_A1 --rg LB:KT_LB_140324_A1 --rg \
2024-09-20 15:25:22,699 paleomix.pipeline ERROR                       PU:Lane_1 --rg PL:ILLUMINA --rg PG:bwa --rg \
2024-09-20 15:25:22,699 paleomix.pipeline ERROR                       'DS:/media/jbod4/kristina_tabbada_jbod4/mus_merged_data/KT_LB_140324_A1_EKDL240005610-1A_22KL5WLT3_L8_[12].fq' \
2024-09-20 15:25:22,700 paleomix.pipeline ERROR                       -q 30 -F 0x4
2024-09-20 15:25:22,700 paleomix.pipeline ERROR         Status  = Automatically terminated by PALEOMIX
2024-09-20 15:25:22,700 paleomix.pipeline ERROR         STDIN   = Piped from process 1
2024-09-20 15:25:22,700 paleomix.pipeline ERROR         STDOUT  = '/media/jbod4/kristina_tabbada_jbod4/mus_merged_data/temp/fdc978d4-0eb1-4f1c-918f-45b159d50d09/collapsed.bam'
2024-09-20 15:25:22,700 paleomix.pipeline ERROR         STDERR* = '/media/jbod4/kristina_tabbada_jbod4/mus_merged_data/temp/fdc978d4-0eb1-4f1c-918f-45b159d50d09/pipe_python_140447126230016.stderr'
2024-09-20 15:25:22,700 paleomix.pipeline ERROR         CWD     = '/media/jbod4/kristina_tabbada_jbod4/mus_merged_data'
2024-09-20 15:25:33,449 paleomix.pipeline ERROR NodeError while alignment of './GD_Boncuklu_mice_reanalysis_linear_o2/reads/KT_LB_140324_A1/KT_LB_140324_A1/Lane_1/KT_LB_140324_A1_EKDL240005610-1A_22KL5WLT3_L8_x.fq/reads.collapsed.truncated.gz' onto GCF_000001635.27_GRCm39_genomic using BWA samse:
2024-09-20 15:25:33,450 paleomix.pipeline ERROR     Error(s) running Node:

**Comment by MikkelSchubert** [comment]
Hi,

This is a bug in how paleomix applies arguments to BWA backtrack; I'll try to release a fix this weekend. 
I believe that you are using the development version of paleomix, is that right?

Best,
Mikkel

**Comment by MikkelSchubert** [comment]
Actually, I believe that you may be using an outdated copy of the master branch/development version of paleomix, since it turns out that I fixed this issue sometime last year. And it shouldn't apply to the 1.3.x versions. If you are using an older version of the master branch/development version, then try to update your copy of paleomix.

Best,
Mikkel

**Comment by KrisTabb** [comment]
Thanks for your quick response! I am running v1.3.7, do you think updating to 1.3.8 get rid of the error? Thanks and best wishes,

**Comment by MikkelSchubert** [comment]
Yes, sorry, I was a bit off: The problem applies to older development versions (i.e. if you just do a source checkout early last year) and versions *before* v1.3.8, since that was when I released a fix. So upgrading to 1.3.8 should solve the problem for you

**Comment by KrisTabb** [comment]
Thanks so much, I'll try that ASAP. Have a great weekend!
